docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "\n",
+    "# Note: this piece of code is a bit weird when read into a notebook, but it makes much\n",
+    "# more sense if run into Google Colab as it uses the new Form feature\n",
+    "DATASET_SIZE = \"10K lines\" #@param [\"10K lines\", \"100K lines\", \"500K lines\", \"1M lines\", \"5M lines\", \"10M lines\", \"15M lines\", \"20M lines\", \"Full Dataset\"] {type: \"string\"}\n",
+    "\n",
+    "if DATASET_SIZE == \"Full Dataset\":\n",
+    "    DS_LINES = 24058263  # whole combined_data_1.txt file - change here if you change reference file\n",
+    "else:\n",
+    "    DS_LINES = int(DATASET_SIZE.replace(\" lines\", \"\").replace(\"K\", \"000\").replace(\"M\", \"000000\"))\n",
+    "\n",
+    "print(\"Selected Number of Lines: {:,}\".format(DS_LINES))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title **Create** the reference data file `movie_views.txt` { display-mode: \"form\" }\n",
+    "\n",
+    "!head -n $DS_LINES $DATASET_FOLDER/combined_data_1.txt > $DATA_FILE \n",
+    "!du -h $DATA_FILE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "⚠️ **WARNING**\n",
+    "\n",
+    "While preparing this notebook, I realised that the `movie_titles.csv` file is not originally encoded as `utf8`, but the `latin-1` which is unfortunately **not supported** in `apache_beam`.\n",
+    "\n",
+    "Therefore, we are going to convert file encoding to be compatible with `apache_beam` for later use."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title Fix `movie_titles.csv` encoding to UTF-8.  {display-mode: \"form\"}\n",
+    "\n",
+    "!cp $DATASET_FOLDER/movie_titles.csv $DATASET_FOLDER/movie_titles_latin1.csv\n",
+    "with open(DATASET_FOLDER/\"movie_titles_latin1.csv\", encoding=\"latin-1\") as mtl, \\\n",
+    "open(DATASET_FOLDER/\"movie_titles.csv\", \"wb\") as mtut8:\n",
+    "    ll = mtl.read()\n",
+    "    mtut8.write(ll.encode(\"utf-8\"))\n",
+    "\n",
+    "!rm $DATASET_FOLDER/movie_titles_latin1.csv"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "✅ Well done! \n",
+    "\n",
+    "The setup of the `Netflix-prize` dataset is now complete! Now it is time to start setting up our execution framework based on **Apache Beam**."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The Execution Framework\n",
+    "\n",
+    "In this section we will define the core main components that will be used throughout the exercises. These components will be based on **Apache Beam**, which consitutes the reference computational framework, and provides the building blocks to define our data workflows.  "
```

Comment:
typo

Reply:
Sorry @dvadym which one? Review of notebooks via Github is indeed very painful! :) 
Trying to recollect all your useful comments.. thanks!

Reply:
Sry, I probably incorrectly located a position in .ipynb file when mapped from Colab. I can't find it now. Never mind about this comment.

 Yeah, reviewing notebooks in Github is painful :(